### PR TITLE
Data access

### PR DESCRIPTION
**Pull request by IamBeaver**
Have created data access layer. Used entity framework core for this task.

### CODE REVIEW

**Comment by anniemt6** [comment]
You should stick to the commit message style. Read the article below: 
https://dev.to/ruanbrandao/how-to-make-good-git-commits-256k

File: Backend/Backend.sln [hunk 1]
```diff
@@ -5,6 +5,8 @@ VisualStudioVersion = 17.1.32210.238
 MinimumVisualStudioVersion = 10.0.40219.1
```
Comment (anniemt6) [inline Backend/Backend.sln:5]: You should stick to the commit message style. Read the article below:
https://dev.to/ruanbrandao/how-to-make-good-git-commits-256k

Comment (anniemt6) [reply Backend/Backend.sln:5]: In addition, describe what you done in your pull request.

File: Backend/DataAccess/Data/DbInitializer.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    public static class DbInitializer
+    {
+        public static async Task Initialize(ApplicationDbContext context)
+        {
+            context.Database.EnsureCreated();
+
+            if (!context.Users.Any())
+            {
+                await context.Users.AddRangeAsync(GetPreconfiguredUsers());
+
```
Comment (anniemt6) [inline Backend/DataAccess/Data/DbInitializer.cs]: Code style. Here is unnecessary spaces.

File: Backend/DataAccess/Data/Configurations/UserEntityTypeConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+﻿using DataAccess.Data.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace DataAccess.Data.Configurations
+{
+    public class UserEntityTypeConfiguration : IEntityTypeConfiguration<User>
```
Comment (anniemt6) [inline Backend/DataAccess/Data/Configurations/UserEntityTypeConfiguration.cs]: It`s better to name UsersEntityTypeConfiguration.

File: Backend/DataAccess/Data/Configurations/UserEntityTypeConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    {
+        public void Configure(EntityTypeBuilder<User> builder)
+        {
+            builder.HasKey(x => x.Id);
+            builder.Property(x => x.Id).UseHiLo("user_hilo").IsRequired();
+            builder.Property(x => x.FirstName).HasMaxLength(30).IsRequired();
+            builder.Property(x => x.LastName).HasMaxLength(30).IsRequired();
+            builder.Property(x => x.Age).IsRequired();
```
Comment (anniemt6) [inline Backend/DataAccess/Data/Configurations/UserEntityTypeConfiguration.cs]: Input values are dublicated. Maybe it's better to insert `HasMaxLength` value in a constant.

File: Backend/DataAccess/Data/DbInitializer.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+                await context.SaveChangesAsync();
+            }
```
Comment (MarikNez) [inline Backend/DataAccess/Data/DbInitializer.cs]: There shouldn't be an empty space between If/else  construction and async methods inside

File: Backend/DataAccess/Data/Entities/User.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace DataAccess.Data.Entities
+{
+    public class User
+    {
+        public int Id { get; set; }
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+
```
Comment (MarikNez) [inline Backend/DataAccess/Data/Entities/User.cs]: Delete empty spce between properties

**Pull request merged**